relay-config/src/config.rs

```diff
@@ -926,7 +926,7 @@ fn spool_max_backpressure_envelopes() -> usize {
 
 /// Default max memory usage for unspooling.
 fn spool_max_backpressure_memory_percent() -> f32 {
-    0.9
+    0.8
```

Comment:
I considered not changing the default and only lowering it for SaaS, but

1. This will only change behavior for users that have spooling configured _and_ run relay at ~80% memory utilization.
2. It might actually benefit self-hosted / managed relay users to switch to more sensible defaults.